src/utils/transaction.ts

```diff
@@ -314,6 +421,308 @@ export const buildCrossChainAction = async (
+            }
+  
+            const result = await plrDaoStaking(offer,BigNumber.from(bestRoute.toAmount).toString(), sdk.state.walletAddress, sdk);
+  
+            if (result.errorMessage) return { errorMessage: result.errorMessage };
+  
+            result.result?.transactions.map((element) => {
+              destinationTxns.push(element);
+            });
+  
+            const preview = {
+              fromChainId,
+              hasEnoughPLR: transactionBlock?.values?.hasEnoughPLR,
+              fromAsset: {
+                address: fromAssetAddress,
+                decimals: fromAssetDecimals,
+                symbol: fromAssetSymbol,
+                amount: amountBN.toString(),
+                iconUrl: fromAssetIconUrl,
+              },
+              amount: BigNumber.from(bestRoute.toAmount),
+              toAsset: {
+                address: plrDaoAsset.address,
+                decimals: 9,
+                symbol: 'PillarDAO',
+                amount: result.result?.returnAmount.toString() ?? '0',
+                iconUrl: 'https://public.pillar.fi/files/pillar-dao-member-badge.png',
+              },
+              receiverAddress: transactionBlock?.values?.receiverAddress,
+              providerName: result.result?.provider ? swapServiceIdToDetails[result.result.provider].title : 'Unknown provider',
+              providerIconUrl: result.result?.provider ? swapServiceIdToDetails[result.result.provider].iconUrl : '',
+            };
+  
+            const crossChainAction: ICrossChainAction = {
+              id: crossChainActionId,
+              relatedTransactionBlockId: transactionBlock.id,
+              chainId: fromChainId,
+              type: TRANSACTION_BLOCK_TYPE.PLR_DAO_STAKE,
+              preview,
+              transactions,
+              isEstimating: false,
+              estimated: null,
+              containsSwitchChain: bestRoute.containsSwitchChain,
+              bridgeUsed: bestRoute.steps[0].tool,
+              receiveAmount: bestRoute.toAmount,
+              useWeb3Provider: accountType === AccountTypes.Key,
+              destinationCrossChainAction: [{
+                id: uniqueId(`${createTimestamp}-`),
+                relatedTransactionBlockId: transactionBlock.id,
+                chainId: plrDaoAsset.chainId,
+                type: TRANSACTION_BLOCK_TYPE.PLR_DAO_STAKE,
+                preview,
+                transactions: destinationTxns,
+                isEstimating: false,
+                estimated: null,
+                useWeb3Provider: false,
+                destinationCrossChainAction: [],
+                gasTokenAddress: plrDaoAsset.address
+              }],
+            };
+  
+            return { crossChainAction };
+          } catch (e) {
+            return { errorMessage: 'Failed to get bridge route!' };
+          }
+  
+      } catch (e) {
+        return { errorMessage: 'Failed to get PLR Dao staking transaction!' };
+      }
+    }
+    
+  }
+
```

Comment:
Ooooff, this is one huge block of nested if / elses. Could you try and split this up into smaller chunks of code that could possibly be reused? As it looks like some code is repeating itself...